### PR TITLE
Add missing config options to formatting rules

### PR DESCRIPTION
**Pull request by 3flex**

### CODE REVIEW

File: detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/wrappers/FunctionName.kt [hunk 1]
```diff
@@ -13,4 +17,12 @@ class FunctionName(config: Config) : FormattingRule(
     "Function name should start with a lowercase letter (except factory methods) and use camel case."
 ) {
     override val wrapping = FunctionNamingRule()
+
+    @Configuration("ignore functions annotated with")
+    private val ignoreWhenAnnotatedWith by config("")
```
Comment (BraisGabin) [inline detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/wrappers/FunctionName.kt:22]: This one is the same as ignoreAnotated that we support by default on all the rules. Does it have sense to give this configuration? Or should we pass an empty string to the ktlint rule and "force" our users to use `ignoreAnotated`

**Review by BraisGabin — APPROVED** [review]
I disabled the auto merge just to know your opinion on this. But I'm not against merging this as it is either.

File: detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/wrappers/FunctionName.kt (hunk 1 above)
Comment (3flex) [reply detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/wrappers/FunctionName.kt:22]: Oh good point - yes I think that makes sense.

**Review by BraisGabin — APPROVED** [review]
LGTM

**Pull request merged**